mailpoet/assets/js/src/settings/store/create_reducer.ts

```diff
@@ -16,17 +16,23 @@ export function createReducer(defaultValue: State) {
     action: Action,
   ): State => {
     switch (action.type) {
-      case 'SET_SETTING':
-        return setWith(
+      case 'SET_SETTING': {
+        const newState = setWith(
           clone(state),
           ['data', ...action.path],
           action.value,
           clone,
         );
+
+        newState.hasUnsavedChanges = true;
```

Comment:
I would consider putting this under `state.save` so `newState.save.hasUnsavedChanges = true;`. I think it is closely related to saving as it holds the info about unsaved changes, and it resets when saving is done. I think it is a good practice (which we didn't always follow) to keep the root of the store transparent.

Reply:
Good point. You had mentioned it in our initial conversation and I completely forgot about it. I just implemented it in a new commit.